types/signed_header.go

```diff
@@ -22,7 +26,41 @@ func (sH *SignedHeader) Verify(untrst header.Header) error {
 		// and is a programmer bug
 		panic(fmt.Errorf("%T is not of type %T", untrst, untrstH))
 	}
+	err := untrstH.ValidateBasic()
```

Comment:
Pulling the verification steps from ValidateBasic() from 'validation.go' into here might make sense so we can verify a type in one place. I suggest we start splitting the types package into its packages. Let's say a header package with dedicated tests for verification of the header, for example. thoughts cc @MSevey

Reply:
If all the code related to headers fits nicely into a single file then I don't think we need to sub divide the packages.  

One types folder with a single file and test file per category is reasonable imo. 
If we we need to start expanding into multiple files to handle all the types code associated with the header for example, then I think it would make more sense to sub divide the package so that you were just pre-pending `header_*.go` to everything.

Reply:
Interesting I suggest we revisit this when we think about a more general refactor for cleaner code.  What's your opinion on my first point ?
> Pulling the verification steps from ValidateBasic() from 'validation.go' into here might make sense so we can verify a type in one place.

Reply:
> pulling the verification steps from ValidateBasic() from 'validation.go' into here might make sense so we can verify a type in one place.

Yes agree, we should think of files as scope barriers. So `signed_header.go` should contain all the code related to signed headers.

Reply:
This honestly can be applied to the rest of the code in the `validation.go` file.

Reply:
Removed `validation.go` and moved all related validation logic to respective files.